app/src/main/java/org/septa/android/app/database/SEPTADatabase.java

```diff
@@ -14,37 +14,73 @@
      */
-    private static final int DATABASE_VERSION = 282;
-    private static final String DATABASE_FILE_NAME = "SEPTA.sqlite";
+    private static final String LATEST_DATABASE_API_URL = "https://s3.amazonaws.com/mobiledb.septa.org/latest/latestDb.json";
+
+    // these are left in case the user does not have the most up to date version of the database
+    // modify databaseVersion and databaseFileName when pushing out a new release with a new DB
+    private static int databaseVersion = 282;
+    private static String databaseFileName = "SEPTA.sqlite";
+    // TODO: have to update shared preferences when these constants are changed so that older DBs don't overwrite a newer one on the app
+
+    public SEPTADatabase(Context context, int databaseVersion, String databaseFileName) {
+        super(context, databaseFileName, null, databaseVersion);
+
+        setDatabaseVersion(databaseVersion);
+
+        CrashlyticsManager.log(Log.INFO, TAG, "Initializing DB: " + databaseVersion);
 
-    public SEPTADatabase(Context context) {
-        super(context, DATABASE_FILE_NAME, null, DATABASE_VERSION);
-        CrashlyticsManager.log(Log.INFO, TAG, "Initializing DB:" + DATABASE_FILE_NAME);
         // Causes database to overwrite when version changes
         setForcedUpgrade();
 
-//        execSQL(new String[]{"CREATE INDEX IF NOT EXISTS tripsBUSIDX on trips_bus(trip_id)"});
-//        execSQL(new String[]{"CREATE INDEX IF NOT EXISTS stopsBUSSTOPID on stop_times_bus(stop_id)"});
-//        execSQL(new String[]{"CREATE INDEX IF NOT EXISTS stopsBUSSSEQ on stop_times_bus(stop_sequence)"});
-//
-//        //execSQL(new String[]{"drop table stop_route_direction"});
-//
-//        execSQL(new String[]{"CREATE TABLE stop_route_direction (stop_id INT, route_id TEXT, direction_id INT)",
-//                "insert into stop_route_direction(stop_id, route_id, direction_id) SELECT DISTINCT S.stop_id, T.route_id, direction_id   FROM trips_bus T join stop_times_bus ST ON T.trip_id = ST.trip_id join stops_bus S on ST.stop_id = S.stop_id",
-//                "insert into stop_route_direction(stop_id, route_id, direction_id) SELECT DISTINCT S.stop_id, T.route_id, direction_id   FROM trips_MFL T join stop_times_MFL ST ON T.trip_id = ST.trip_id join stops_bus S on ST.stop_id = S.stop_id",
-//                "insert into stop_route_direction(stop_id, route_id, direction_id) SELECT DISTINCT S.stop_id, T.route_id, direction_id   FROM trips_BSL T join stop_times_BSL ST ON T.trip_id = ST.trip_id join stops_bus S on ST.stop_id = S.stop_id",
-//                "CREATE INDEX stop_srd_index ON stop_route_direction (stop_id)",
-//                "CREATE INDEX route_srd_index ON stop_route_direction (route_id)",
-//                "CREATE INDEX direction_srd_index ON stop_route_direction (direction_id)"});
-
+        // create indices on new app database
+        execSQL(new String[]{
+                "CREATE INDEX stop_srd_index ON stop_route_direction (stop_id);",
```

Comment:
These should be externalized.  You can store them in a text file that you read in or you can put them into the Strings values with the other SQL.  But hard coding them is not a good idea.

Reply:
You can also execute them all in 1 shot if they are separated by ";"